addons/website_slides/controllers/main.py

```diff
@@ -301,17 +301,40 @@ def create_slide(self, *args, **post):
     # --------------------------------------------------
     @http.route('/slides/embed/<int:slide_id>', type='http', auth='public', website=True, sitemap=False)
     def slides_embed(self, slide_id, page="1", **kw):
+        """
+        Detect if the slide is embedded in a website outside of the Odoo database.
+        The detection is there to differentiate cases where the slide is present on a webpage that is different than
+        the website for which the slide was created
+
+        This is how we decide if the slide is embedded or not:
+        if referer_url is Falsly -> Not embedded (it's the default behavior)
+
+        if there is a website_id (which means the slide is just available for 1 website) ->
+            is_embedded = False  if and only if  domain of referer is equal to domain of website
+        else (Available for all websites) ->
+            is_embedded = False  if and only if  domain of referer is equal to ANY of the domain of the odoo DB
+
+
+        This is important because externally embedding the slide will display Share options inside the slide itself,
+        rather than on the bottom of the page in the Odoo slide module
+
+        In case of uncertainty about the situation, we should assume it is NOT embedded
+        """
         # Note : don't use the 'model' in the route (use 'slide_id'), otherwise if public cannot access the embedded
         # slide, the error will be the website.403 page instead of the one of the website_slides.embed_slide.
         # Do not forget the rendering here will be displayed in the embedded iframe
 
         # determine if it is embedded from external web page
-        referrer_url = request.httprequest.headers.get('Referer', '')
-        base_url = request.env['ir.config_parameter'].sudo().get_param('web.base.url')
-        is_embedded = referrer_url and not bool(base_url in referrer_url) or False
-        # try accessing slide, and display to corresponding template
+        referrer_url = request.httprequest.headers.get('Referer', False)
+        referer_domain = tools.url_domain_extract(referrer_url)
         try:
             slide = request.env['slide.slide'].browse(slide_id)
+            website_id = slide.website_id
+            if website_id:
+                is_embedded = referer_domain and not referer_domain.endswith(website_id._get_http_domain())
+            else:
+                is_embedded = referer_domain and not any(referer_domain.endswith(web_id._get_http_domain()) for web_id
+                                                         in request.env['website'].search([('domain', '!=', '')]))
```

Comment:
Isn't it False the default void value for a char in a search ?

Reply:
Changed `([('domain', '!=', '')]))` to `([('domain', '!=', False)]))`